ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V100__Append_only_schema.sql

```diff
@@ -0,0 +1,753 @@
+    participant_events.exercise_argument,
+    participant_events.exercise_argument_compression,
+    participant_events.exercise_result,
+    participant_events.exercise_result_compression,
+    participant_events.exercise_actors,
+    participant_events.exercise_child_event_ids
+FROM participant_events
+WHERE participant_events.exercise_consuming = FALSE -- non-consuming exercise events
+ORDER BY participant_events.event_sequential_id;
+
+-- Temporary sequence and table used to generate sequential IDs that appear after
+-- all other already existing sequential IDs.
+-- We are using plain temporary objects to minimize the amount of database-specific SQL features used.
+CREATE SEQUENCE temp_divulgence_sequential_id START 1;
+CREATE TABLE temp_ledger_end_sequential_info (
+    ledger_end_sequential_id bigint NOT NULL
+);
+INSERT INTO temp_ledger_end_sequential_info(
+    SELECT coalesce(max(event_sequential_id), 0) FROM participant_events
+);
+
+-- A temporary table with data for existing divulgence events.
+-- Divulgence events did not exist before, we need to assign a new sequential ID for them.
+-- They will all be inserted after all other events, i.e., at a point later than the transaction
+-- that actually lead to the divulgence. This is OK, as we only use the for lookups
+-- AND we can recognize these divulgence events from not having an associated ???.
+-- In addition, we want to avoid rewriting the event_sequential_id of other events
+-- for data continuity reasons.
+CREATE TABLE temp_divulged_contracts (
+    event_sequential_id bigint NOT NULL,
+    contract_id text NOT NULL,
+    divulgees text[] NOT NULL
+);
+INSERT INTO temp_divulged_contracts (
+    event_sequential_id,
+    contract_id,
+    divulgees
+)
+SELECT
+    nextval('temp_divulgence_sequential_id') + ledger_end_sequential_id,
+    contract_id,
+    array_agg(contract_witness)
+FROM
+    participant_contract_witnesses
+        INNER JOIN participant_contracts USING (contract_id)
+        LEFT JOIN participant_events USING (contract_id),
+    temp_ledger_end_sequential_info
+WHERE (NOT create_stakeholders @> array[contract_witness])
+    AND exercise_consuming IS NULL -- create events only
+    AND (tree_event_witnesses IS NULL OR NOT tree_event_witnesses @> array[contract_witness])
+GROUP BY contract_id, ledger_end_sequential_id;
+
+INSERT INTO participant_events_divulgence (
+    event_sequential_id,
+    event_offset,
+    command_id,
+    workflow_id,
+    application_id,
+    submitters,
+    contract_id,
+    template_id,
+    tree_event_witnesses,
+    create_argument,
+    create_argument_compression
+)
+SELECT
+    event_sequential_id,
+    -- The following 5 fields are metadata of the transaction that lead to the divulgence.
+    -- We can't reconstruct this information from the old schema.
+    NULL, -- TODO append-only: this is the only reason why the event_offset is nullable. Use some non-null offset instead?
+    NULL,
+    'flyway-migration',
```

Comment:
Good point. This is the workflow ID, which is normally visible through the active contract service and the transaction service. 

I don't think we serve divulged contracts through any service right now. Once we start serving them, this string might be visible to users (although it would seem strange to me to expose transaction metadata to users to which the contract was only divulged to).

In any case, AFAICS we can't reconstruct the original workflow ID here, so we have to invent some new ID.

Reply:
What about setting it to the empty string, which is the default used by the Ledger API server when no workflow ID is given?

This way there would be less confusion if we accidentally serve the workflow ID.